test/offline_queue_test.exs

```diff
@@ -0,0 +1,153 @@
+  end
+
+  test "Check updating stream format and scene on send buffer" do
+    state = setup_videos()
+
+    actions = pad1_actions()
+    {^actions, _state} = OfflineQueue.handle_process(@pad1, send_buffer(0), %{}, state)
+  end
+
+  test "Check removed pad doesn't block queue and if stream format, scene and buffer actions are send" do
+    state = setup_videos()
+    assert {[], state} = OfflineQueue.handle_process(@pad2, send_buffer(0), {}, state)
+    assert {[], state} = OfflineQueue.handle_process(@pad2, send_buffer(1_000_000_000), {}, state)
+
+    assert {_pad1_buffer1_actions, state} =
+             OfflineQueue.handle_process(@pad1, send_buffer(0), %{}, state)
+
+    pad1_buffer2_action = get_buffer_action(@pad1, 1_000_000_000)
+
+    assert {^pad1_buffer2_action, state} =
+             OfflineQueue.handle_process(@pad1, send_buffer(1_000_000_000), %{}, state)
+
+    second_pad_actions = unlocked_pad2_actions()
+    assert {^second_pad_actions, _state} = OfflineQueue.handle_pad_removed(@pad1, %{}, state)
+  end
+
+  test "Check update scene messages handling" do
+    state = setup_videos()
+
+    {[], state} = OfflineQueue.handle_process(@pad2, send_buffer(0), %{}, state)
+    {[], state} = OfflineQueue.handle_process(@pad2, send_buffer(1_000_000_000), %{}, state)
+
+    new_scene = %Scene{videos_configs: %{}}
+    {[], state} = OfflineQueue.handle_parent_notification({:update_scene, new_scene}, %{}, state)
+
+    assert {_pad1_buffer1_actions, state} =
+             OfflineQueue.handle_process(@pad1, send_buffer(0), %{}, state)
+
+    assert {_pad1_buffer2_actions, state} =
+             OfflineQueue.handle_process(@pad1, send_buffer(1_000_000_000), %{}, state)
+
+    {second_pad_actions, _state} = OfflineQueue.handle_pad_removed(@pad1, %{}, state)
+
+    scene_update_action = {:notify_child, {:compositor_core, {:update_scene, new_scene}}}
+    assert ^scene_update_action = Enum.at(second_pad_actions, -2)
+  end
+
+  defp send_buffer(pts) do
+    %Buffer{
+      pts: pts,
+      dts: pts,
+      payload: @frame
+    }
+  end
+
+  defp setup_videos() do
+    pad1_options = %{video_config: @video_config, timestamp_offset: 0}
+    pad2_options = %{video_config: @video_config, timestamp_offset: @pad2_pts_offset}
+
+    assert {[], state} = OfflineQueue.handle_init(%{}, %{target_fps: {1, 1}})
+
+    assert {[], state} = OfflineQueue.handle_pad_added(@pad1, %{options: pad1_options}, state)
+
+    assert {[], state} =
+             OfflineQueue.handle_stream_format(@pad1, @video_stream_format, %{}, state)
+
+    assert {[], state} = OfflineQueue.handle_pad_added(@pad2, %{options: pad2_options}, state)
+
+    assert {[], state} =
+             OfflineQueue.handle_stream_format(@pad2, @video_stream_format, %{}, state)
+
+    state
+  end
+
+  @spec pad1_actions() :: [
+          Action.stream_format_t() | [State.notify_compositor_scene() | Action.buffer_t()]
+        ]
+  defp pad1_actions() do
+    stream_format_action = [
```

Comment:
[NIT] The variable is called `stream_format_action`, and its value is a list (one-element list, but still a list :D) - perhaps it would be better to simply return `[stream_format_action, scene_action, buffer_action]` from that function, with `*_action` defined as elements, rather than lists?